### DISCUSSION TITLE
General setup and correcting mistakes in setup

### DISCUSSION THREAD

**Discussion by imnotjack91** [immense/Remotely#196] [Q&A]
hey all, 
so i negleted to use the FQDN during the powershell setup script and used the server name instead. im assuming that script requires all 4-5 inputs to be the FQDN? 
and can i just run the setup again to overwrite the previous settings or uninstall/reinstall? or would it be easieer to delete and recreate this vm?

**Comment by bitbound** [comment]
The script contains comments about what the parameters mean.

![image](https://user-images.githubusercontent.com/20995508/103069139-94817600-4573-11eb-9265-5dad64e40599.png)

If you're unfamiliar with hosting an ASP.NET Core app on IIS, I recommend going over the Microsoft documentation:  https://docs.microsoft.com/en-us/aspnet/core/tutorials/publish-to-iis?view=aspnetcore-5.0&tabs=visual-studio

**Reply by imnotjack91** [reply]
so i got ssl configured and i got the server to give me the 503 error. i feel like its "working" now. like routing traffic correctly. 
i also just realized ive been passed back and forth around the readme file not realizing that existed.
im going to give that a read and see if my now perpetual headache relieves itself

**Comment by bitbound** [comment]
I remember having some issues when I first set up my Windows server too.  But after the first hiccup, I've never had any issues.  There's no reverse proxy to worry about, and it seems to generally outperform Linux.  And MS Deploy is nice.

Anyhow, don't get discouraged.  Once it's working, it'll stay working.

Here are some things you can try.

Reboot if you haven't since installing the .NET 5.

Go into the site folder (e.g. C:\inetpub\\{sitename}).  Find `web.config` file.  Open it and change `stdoutLogEnabled="false"` to `stdoutLogEnabled="true" `.  Restart the site, then try to browse to it again.  You should now have logs in C:\inetpub\\{sitename}\logs that might be able to tell you what's crashing it.

If the logs say something about the AspNetCoreModule, try reinstalling .NET 5 hosting bundle and rebooting.

Remember to turn off the above stdoutLogEnabled when you're done.

If you're still getting the error about the module, find the file `C:\Windows\System32\inetsrv\Config\applicationHost.config`.  **Make a copy of it before you do anything.**  Then search for all instances of `AspNetCoreModule`.  Delete the entire line for everywhere you find it.  Reinstall .NET 5 hosting bundle and reboot.

**Reply by imnotjack91** [reply]
so there appears to be nothing in inetpub 
![image](https://user-images.githubusercontent.com/62719825/103318047-be62ee80-49f2-11eb-90c6-9dfff7dbaa0c.png)

**Reply by imnotjack91** [reply]
i did find that the directory i ran the poweshell script from now has a folder the name of the site i intended to make. do i need to copy that to the inetpub folder?

**Reply by imnotjack91** [reply]
also the web.config file does not appear to be present

**Comment by imnotjack91** [comment]
ok, it is painfully clear i cannot resolve my issues by myself. 
i will admit, this might be a bit much for a beginner project as an into into IIS projects. 
so heres where all my trials and tribulations have left me.
it is clear the script does some of  the setup for me in IIS. its unclear to me if i need to do anything else before or after running the script. there does not seem to be any further instructions regarding the powershell script.
so here are my questions.
1. Where does the script need to be run from, im assuming the downloads folder is not the right answer
2. are there any steps other than installing .net and IIS bundle before i run the script? (including where to run the scriipt from)
3. where should i make the path to the site point to? (does it matter, will IIS know where it is if its not in the IIS directory, should i put it in a virtual drive attatched to the VM? is it easier for a beginner to put it in the IIS folder, what is best practices for the inexperienced here?)
4. are there any steps to take AFTER running the script?
5. do i need to get the admin account created BEFORE i enable SSL or does it matter?
6. The instructions say to build the app before starting on the local steps, is this still necessary when using the build script?

THANK YOU for developing this application. i would sponsor you if i had the money, if i make money with this, i will see if i can set something up.

Thank you for your help and advise. im super grateful for all the help you have offered this community.
